### PR TITLE
Fix numerical error in #12

### PR DESCRIPTION
**Pull request by mohamed82008**
This PR implements the solution described in https://github.com/TuringLang/Bijectors.jl/issues/12#issuecomment-453760477. I also relaxed the tolerance for the inverse check in case of the Dirichlet distribution for the reason explained in the comments.

### CODE REVIEW

**Comment by mohamed82008** [comment]
Also fixes #8.

**Comment by mohamed82008** [comment]
Supersedes #13.

**Comment by yebai** [comment]
Thanks, @mohamed82008. I'll leave this open to get some further feedback. 

In the meantime, let's try to apply a similar fix to bounded univariate distributions.

**Comment by mohamed82008** [comment]
This is ready for a review.

File: src/Bijectors.jl [hunk 1]
```diff
@@ -60,8 +60,12 @@ end
 #############
 
 const TransformDistribution{T<:ContinuousUnivariateDistribution} = Union{T, Truncated{T}}
+@inline function _clamp(x::Real, dist::TransformDistribution)
+    return clamp(x, minimum(dist), maximum(dist))
```
[the remark this answers is not in the document]
Comment (xukai92) [reply src/Bijectors.jl]: Probably use `@debug`? Otherwise frequent calls will give too many messages.

**Comment by yebai** [comment]
Thanks, @mohamed82008!

**Pull request merged**